tests/conftest.py

```diff
@@ -10,7 +11,12 @@
 from tests.testutils import tempfile
 
 
-def pytest_addoption(parser):
+if TYPE_CHECKING:
```

Comment:
you shouldn't need the `if TYPE_CHECKING` pattern here- that's really only for preventing circular imports. You can just import these directly, and use them below without quoting the types

Reply:
That's a convention I'm used to. If you need a module just for type hints, import it only in the `if TYPE_CHECKING` block. 

It's easier to teach to other people ("Why are you sometime put import in that block and sometimes not?") and the overall chance that you have a circular import if you need this import just for type hint is quite big.

TL;DR: I prefer to leave it as it is.

Reply:
fair enough. there's some interesting background on it here - https://github.com/larryhastings/co_annotations/issues/1

in general, it seems that `if TYPE_CHECKING` is used iff

- you would otherwise have a circular import
- the import is 'expensive'

the Benevolent Dictator for Life seems to regard it is a painfully necessary hack.

that said, it's also clear from the thread that it's far from settled. And using it *everywhere* instead of the small number of cases where you need it (do you definitely need it anywhere in this codebase?) is at least consistent.